src/electron/process_manager.ts

```diff
@@ -272,100 +176,130 @@ export class ConnectionManager {
   }
 }
 
-// Simple "one shot" child process launcher.
-//
-// NOTE: Because there is no way in Node.js to tell whether a process launched successfully,
-//       #startInternal always succeeds; use #onExit to be notified when the process has exited
-//       (which may be immediately after calling #startInternal if, e.g. the binary cannot be
-//       found).
-class ChildProcessHelper {
-  private process?: ChildProcess;
-
-  private exitListener?: () => void;
+// Class to manage the lifecycle of tun2socks.
+class Tun2socks {
+  private process: ChildProcessHelper;
+  private isUdpEnabled = false;
 
-  constructor(private path: string) {}
+  private notifyStop = true;
```

Comment:
I'm getting a little confused trying to understand this `notifyStop` thing, and I worry that it is error-prone.  For example, if `notifyStop` becomes false (i.e. the process is restarted), and then the child process crashes without printing anything, `Tun2socks.stopped` will never resolve or reject.

Instead of having `notifyStopped` here, I think a better solution might be to make`ChildProcessHelper.stop` replace the exit listener (and be async to report successful shutdown), and have `ChildProcessHelper.onExit` only resolve on spontaneous exits.

Reply:
Good suggestion. Done.